contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -117,4 +121,8 @@ contract TransparentUpgradeableProxy is ERC1967Proxy {
         require(msg.sender != _getAdmin(), "TransparentUpgradeableProxy: admin cannot fallback to proxy target");
         super._beforeFallback();
     }
+
+    function _nonPayable() private {
```

Comment:
This issue is pretty esoteric so we should add a good comment here explaining why it's needed.

I would also consider a different name for this function. Something like `_requireZeroValue()`?

Reply:
done